### PR TITLE
Changed to Mousetrap

### PR DESCRIPTION
**Pull request by tkroll**
Hello again,

Redid the fork, removed the old PRs, and made this one off current master.

* Removed Keymaster and changed to Mousetrap
* Updated README
* Bumped to 0.3.4

### CODE REVIEW

**Comment by tkroll** [comment]
Also added an option to hide the selected search result outline

File: src/options.js [hunk 1]
```diff
@@ -9,101 +9,70 @@ const flashMessage = message => {
   }, 3000);
 };
 
+const divs = {
```
Comment (infokiller) [inline src/options.js]: Please use a more descriptive name, perhaps `divToOptionName`

**Comment by infokiller** [comment]
Thanks @tkroll !
Added some comments inline. In addition, I have two general style comments:
1. Start comments with an uppercase letter and end them with a dot.
2. Format the code using prettier

File: src/extension.js [hunk 2]
```diff
@@ -5,32 +5,32 @@ const extension = {
...
+      navigateFlightsTab: ['alt+l'],
+      navigateFinancialTab: ['f'],
+      focusSearchInput: ['/', 'escape'],
+      navigateShowAll: ['z z'],
+      navigateShowHour: ['z h'],
+      navigateShowDay: ['z d'],
+      navigateShowWeek: ['z w'],
+      navigateShowMonth: ['z m'],
+      navigateShowYear: ['z y'],
+      toggleSort: ['z s']
```
[the remark this answers is not in the document]
Comment (tkroll) [reply src/extension.js]: Mousetrap want's an array. I handle the conversion between string and array from the options. It works the same way as before.

File: src/main.js [hunk 3]
```diff
@@ -161,7 +141,7 @@ Object.assign(extension, {
         // but it won't remove the focus
         return true;
       }
-    }, this.scopes.searchbox);
```
[the remark this answers is not in the document]
Comment (tkroll) [reply src/main.js]: Will check

File: src/options.js (hunk 1 above)
Comment (tkroll) [reply src/options.js]: Yes mom :)

File: src/main.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (tkroll) [reply src/main.js]: Yes, same behavior.

**Comment by tkroll** [comment]
Good afternoon. I made the recommended changes and pushed to my branch. Do I need to do something else?

**Comment by tkroll** [comment]
Ready for you to take a look at these updates.

**Comment by mancze** [comment]
It took me some time to discover why migrating to `mousetrap`. The reason is ["chord" support](https://github.com/infokiller/web-search-navigator/issues/30#issuecomment-397570163). Just wanna note it here.

Regarding review: I'm really busy lately (not only) so I cannot review PR properly, sorry. Eventually it should get better but I don't want to promise anything.

**Comment by infokiller** [comment]
@tkroll sorry for the delay, I was also busy lately. Will try to take a look now.

File: src/extension.js [hunk 4]
```diff
@@ -5,32 +5,32 @@ const extension = {
...
+      navigateImagesTab: ['i'],
+      navigateVideosTab: ['v'],
+      navigateMapsTab: ['m'],
+      navigateNewsTab: ['n'],
+      navigateShoppingTab: ['alt+s'],
+      navigateBooksTab: ['b'],
+      navigateFlightsTab: ['alt+l'],
+      navigateFinancialTab: ['f'],
+      focusSearchInput: ['/', 'escape'],
+      navigateShowAll: ['z z', 'command+shift+a'],
```
Comment (infokiller) [inline src/extension.js]: Shouldn't this and the others below include both `command` and `ctrl` shortcuts?

File: src/search_results.css [hunk 5]
```diff
@@ -1,10 +1,15 @@
...
+}
+
+.highlighted-search-result::before,
+.unhighlighted-search-result::before {
   content: "\25BA";
   margin-right: 25px;
   left: -25px;
   position: absolute;
-}
+}
```
Comment (infokiller) [inline src/search_results.css]: I think `highlighted-search-result` and `unhighlighted-search-result` is a bit confusing. Perhaps rename the former to `focused-search-result` and the latter to `no-outline`?
Then always add the `focused-search-result` class to the focused element, and conditionally add the `no-outline` class if the option is set.

File: src/options.js [hunk 6]
```diff
@@ -9,101 +9,74 @@ const flashMessage = message => {
...
+    document.getElementById('wrap-navigation').checked = extension.options.sync.values.wrapNavigation;
+    document.getElementById('auto-select-first').checked = extension.options.sync.values.autoSelectFirst;
+    document.getElementById('hide-outline').checked = extension.options.sync.values.hideOutline;
+
+    // Restore options from divs. 
+    for(let key in divToOptionName) {
+
+      // Options are stored as arrays.
+      // Split them into comma-separated string for the uset.
+      document.getElementById(divToOptionName[key]).value = extension.options.sync.values[key].join(', ')
```
Comment (infokiller) [inline src/options.js]: If a user customized their keybindings, `extension.options.sync.values[key]` will return a string, and the `join` call will fail. That's what I was referring to in the comment about string/array backwards compatibility.

**Comment by infokiller** [comment]
Thanks, Looking good!
Added more comments. Once you address them I think it should be good for a merge, though I will need to test this further before publishing to the store (especially the search field behavior that @mancze fixed, which I wasn't able to reproduce when it was buggy).

**Comment by tkroll** [comment]
Updated requested changes.

**Comment by infokiller** [comment]
Thanks, merging!

**Comment by infokiller** [comment]
OK, seems like you need to resolve conflicts first before I can merge

**Comment by infokiller** [comment]
I think you need to merge or rebase with the current master branch.

**Comment by tkroll** [comment]
Rebased and pushed

**Comment by infokiller** [comment]
I still see that there are conflicts:

image

**Comment by tkroll** [comment]
I'm at a loss. I only removed keymaster and replaced it with mousetrap in package.json. package-lock.json is in .gitignore. I don't have access to the "Resolve Conflicts" button.

**Comment by infokiller** [comment]
It doesn't look like you've actually rebased/merged the current master branch- when I look at tkroll:master it shows that its 12 commits behind infokiller:master.

**Comment by tkroll** [comment]
Hallelujah! Can you tell it's been a while?

**Comment by mancze** [comment]
I know I'm coming a little bit late, but since this is the correct place to ask:

I've noticed that `package-lock.json` file was removed in this PR. Was it only because it was in `.gitignore` file? I think it should have been done the other way around because `package-lock.json` is meant for version control: https://github.com/npm/cli/blob/latest/doc/files/package-lock.json.md .

**Comment by infokiller** [comment]
I'm not sure, but since we use yarn which uses its own lock file, do we really need it?

**Comment by mancze** [comment]
I see, thanks for pointing that out. I forgot that those two actually use different lock files. After some searching it seems that having only single lock file is the best option for this project.

**Pull request merged**